File: elixir/lasagna/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven() do
    40
  end

  def remaining_minutes_in_oven(time_in_the_oven) do
    expected_minutes_in_oven() - time_in_the_oven
  end

  def preparation_time_in_minutes(number_of_layers) do
    number_of_layers * 2
  end

  def total_time_in_minutes(number_of_layers, time_in_the_oven) do
    preparation_time_in_minutes(number_of_layers) + time_in_the_oven
  end

  def alarm do
    "Ding!"
  end
end

File: java/lasagna/src/main/java/Lasagna.java
public class Lasagna {

    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int timeInOven) {
        return this.expectedMinutesInOven() - timeInOven;
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * 2;
    }

    public int totalTimeInMinutes(int numberOfLayers, int timeInOven) {
        return preparationTimeInMinutes(numberOfLayers) + timeInOven;
    }
}
